packages/ra-core/src/controller/usePagination.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
```

Comment:
I'd call this hook `usePaginationState`, because `usePagination` is too generic IMO. Same for `useSort`